app/scripts/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,175 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  const sjcl = require('sjcl');
+
+  function importFxaCryptoDeriver() {
+    return import(/* webpackChunkName: "fxaCryptoDeriver" */ 'fxaCryptoDeriver');
+  }
+
+  const RECOVERY_KEY_VERSION = '\x01';
+  const ALPHABET_BASE32 = '0123456789ABCDEFGHJKMNPQRSTVWXYZ';
+
+  return {
+    /**
+     * Generate a random base32 string of length and
+     * prepend encryption version number.
+     *
+     * @param {Integer} length Length of string to generate
+     * @returns {string}
+     */
+    generateRecoveryKey: function (length) {
+      return importFxaCryptoDeriver().then((fxaCryptoDeriver) => {
+        const jose = fxaCryptoDeriver.jose;
+        const bytes = jose.util.randomBytes(length - 1);
+        const code = [];
+        for (let i = 0; i < length; i++) {
+          code.push(ALPHABET_BASE32[bytes[i] % 32]);
```

Comment:
I'd be curious to know if this introduces bias.  I don't think it should with a good DRBG which attempts uniform distribution, plus it's only 3 bits of loss.

Reply:
I was mistaken; this does introduce a small bias toward the first three characters in the alphabet.  To avoid that bias, this should be masking off those 3 bits instead instead:

```
code.push(ALPHABET_BASE32[bytes[i] & 0x1f);
```

Reply:
Heh, we've been having similar discussions in https://github.com/mozilla/fxa-auth-db-mysql/issues/367 and a private bugzilla bug.  I thought this should be OK because `256 % ALPHABET_BASE32.length` is zero, but am no expert in these things.

Reply:
😄 

I admit the bias is small, maybe small enough that it doesn't increase the attack chances a significant amount.

Reply:
If it's easy to fix we should definitely just do it

Reply:
Since this is a recurring theme, can we extract the base32 code creation logic into one place and re-use it everywhere?